rmgpy/molecule/pathfinder.py

```diff
@@ -256,67 +255,207 @@ def findAllDelocalizationPaths(atom1):
+                            or (atom3.isSulfur() and atom3.lonePairs in [0, 1, 2]))
+                            and bond12.isDouble() and bond23.isDouble()):
+                        paths.append([atom1, atom2, atom3, bond12, bond23])
+    return paths
+
+
+def find_N5dc_delocalization_paths(atom1):
+    """
+    Find all the resonance structures of an N5dc nitrogen atom with a single bond to a radical N/O/S site, another
+    single bond to a negatively charged N/O/S site, and one double bond (not participating in this transformation)
+    Example:
+    - N=[N+]([O])([O-]) <=> N=[N+]([O-])([O]), these structures are isomorphic but not identical, the transition is
+    important for correct degeneracy calculations
+    In this transition atom1 is the middle N+ (N5dc), atom2 is the radical site, and atom3 is negatively charged
+    A "if atom1.atomType.label == 'N5dc'" check should be done before calling this function
+    """
+    cython.declare(paths=list, atom2=Atom, atom3=Atom, bond12=Bond, bond23=Bond)
+
+    paths = []
+
     for atom2, bond12 in atom1.edges.items():
-        # Only triple bonds are considered
-        if bond12.isTriple() and not atom2.isCarbon():
-            for atom3, bond13 in atom1.edges.items():
-                # Only single bonds are considered, at the moment we only consider negatively charged nitrogen and oxygen
-                if (bond13.isSingle() and ((atom3.isNitrogen() and atom3.lonePairs >= 2) or (atom3.isOxygen()
-                            and atom3.lonePairs >= 3))):
-                    paths.append([atom1, atom2, atom3, bond12, bond13, 2])
-    
-    return paths    
+        if atom2.radicalElectrons and bond12.isSingle() and not atom2.charge and is_NOS_able_to_gain_lone_pair(atom2):
+            for atom3, bond23 in atom1.edges.items():
+                if (atom2 is not atom3 and bond23.isSingle() and atom3.charge < 0
+                        and is_NOS_able_to_loose_lone_pair(atom3)):
+                    paths.append([atom2, atom3])
+                    return paths  # there could only be one such path per atom1, return if found
+    return paths
+
+
+def is_NOS_able_to_gain_lone_pair(atom):
+    """
+    Helper function
+    Return True if atom is N/O/S and is able to <gain> an additional lone pair, False otherwise
+    We don't allow O to be with no lone pairs
+    """
+    if (((atom.isNitrogen() or atom.isSulfur()) and atom.lonePairs in [0, 1, 2])
+                        or (atom.isOxygen() and atom.lonePairs in [1, 2])):
+        return True
+    return False
+
+
+def is_NOS_able_to_loose_lone_pair(atom):
+    """
+    Helper function
+    Return True if atom is N/O/S and is able to <loose> a lone pair, False otherwise
+    We don't allow O to remain with no lone pairs
+    """
+    if (((atom.isNitrogen() or atom.isSulfur()) and atom.lonePairs in [1, 2, 3])
+                        or (atom.isOxygen() and atom.lonePairs in [2, 3])):
+        return True
+    return False
```

Comment:
You could directly return the value of your boolean expression (also for gain_lone_pair).

Also, `loose` should be `lose`.